azkaban-common/src/main/java/azkaban/utils/SystemMemoryInfo.java

```diff
@@ -55,9 +55,9 @@ public static void init(int memCheckInterval) {
    * Given Xms/Xmx values (in kb) used by java process, determine if system can
    * satisfy the memory request
    */
-  public synchronized static boolean canSystemGrantMemory(long xms, long xmx, long freeMemDecrAmt) {
+  public synchronized static boolean canSystemGrantMemory(long xms, long xmx, long freeMemDecrAmt) throws Exception {
     if (!memCheckEnabled) {
-      return true;
+      throw new Exception("memory check is disabled");
```

Comment:
If the check is turned off, it seems reasonable to log that the memory is always granted without an exception.

In any case, I would suggest keeping non essential changes and refactoring changes separate.